backend/main.go

```diff
@@ -64,20 +72,73 @@ func init() {
 	}
 }
 
+func newCosmosDBClient(ctx context.Context) (database.DBClient, error) {
+	azcoreClientOptions := azcore.ClientOptions{
+		// FIXME Cloud should be determined by other means.
+		Cloud: cloud.AzurePublic,
+	}
+
+	credential, err := azidentity.NewDefaultAzureCredential(
+		&azidentity.DefaultAzureCredentialOptions{
+			ClientOptions: azcoreClientOptions,
+		})
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := azcosmos.NewClient(argCosmosURL, credential,
+		&azcosmos.ClientOptions{
+			ClientOptions: azcoreClientOptions,
+		})
+	if err != nil {
+		return nil, err
+	}
+
+	databaseClient, err := client.NewDatabase(argCosmosName)
+	if err != nil {
+		return nil, err
+	}
+
+	return database.NewCosmosDBClient(ctx, databaseClient)
+}
+
 func Run(cmd *cobra.Command, args []string) error {
+	ctx := context.Background()
+
 	handler := slog.NewJSONHandler(os.Stdout, nil)
 	logger := slog.New(handler)
 
+	// Create database client
+	dbClient, err := newCosmosDBClient(ctx)
+	if err != nil {
+		return fmt.Errorf("Failed to create database client: %w", err)
+	}
+
+	// Create OCM connection
+	ocmConnection, err := ocmsdk.NewUnauthenticatedConnectionBuilder().
+		URL(argClustersServiceURL).
+		Insecure(argInsecure).
+		Build()
+	if err != nil {
+		return fmt.Errorf("Failed to create OCM connection: %w", err)
+	}
+
 	logger.Info(fmt.Sprintf("%s (%s) started", cmd.Short, cmd.Version))
 
+	operationsScanner := NewOperationsScanner(dbClient, ocmConnection)
+
 	stop := make(chan struct{})
 	signalChannel := make(chan os.Signal, 1)
 	signal.Notify(signalChannel, syscall.SIGINT, syscall.SIGTERM)
 
+	go operationsScanner.Run(ctx, logger, stop)
+
 	sig := <-signalChannel
 	logger.Info(fmt.Sprintf("caught %s signal", sig))
 	close(stop)
 
+	operationsScanner.Join()
```

Comment:
What's the benefit of making `Run()` asynchronous and having a `Join()` as opposed to just making `Run()` synchronous and waiting for it to return?

Reply:
There will eventually be at least one other "scanner" routine running in parallel here so they need to remain asynchronous.

Reply:
If that's true, then this model as written in this PR will not work - one will block you from waiting on the other and you could deadlock. Prefer a `sync.WaitGroup` or `sync.ErrorGroup`